src/utils/motions.ts

```diff
@@ -0,0 +1,55 @@
+import { AnyColonyClient, Extension } from '@colony/colony-js';
+import { TransactionDescription } from 'ethers/lib/utils';
+
+import { mutate } from '../amplifyClient';
+import { ContractEvent, motionNameMapping } from '../types';
+import { getDomainDatabaseId } from './domains';
+import { verbose } from './logger';
+import { getColonyTokenAddress } from './tokens';
+
+export const getParsedActionFromMotion = async (
+  motionId: string,
+  colonyClient: AnyColonyClient,
+) => {
+  const votingClient = await colonyClient.getExtensionClient(
+    Extension.VotingReputation,
+  );
+
+  const motion = await votingClient.getMotion(motionId);
+
+  try {
+    return colonyClient.interface.parseTransaction({
+      data: motion.action,
+    });
+  } catch {
+    verbose(`Unable to parse ${motion.action}`);
+    return undefined;
+  }
+};
+
+export const writeMintTokensMotionToDB = async (
+  {
+    transactionHash,
+    contractAddress: colonyAddress,
+    blockNumber,
+    args: { creator, domainId },
+  }: ContractEvent,
+  parsedAction: TransactionDescription,
+) => {
+  const { name, args: actionArgs } = parsedAction;
+  const amount = actionArgs[0].toString();
```

Comment:
Could this element of the array be null? If so, consider adding optional chaining

Reply:
@jakubcolony No, don't expect so. The front end should not be passing through an undefined amount of tokens to mint.

It maps to the motion params field in the current set up, for reference:

![motionparams](https://user-images.githubusercontent.com/64402732/225981178-8890fb86-a1c7-4488-96cb-19af62692cac.png)

Reply:
Is that a safe assumption to make, that it will never be broken by anyone?
What about calling it directly from a contract?

Reply:
@jakubcolony I don't believe calling it directly from a contract will be a problem either. You need to pass an "encoded action" to the `createMotion` call, and in order for the `encode action` function to succeed, it must receive a complete set of arguments to the underlying action call (by 'complete' I mean correct length, each of which is the correct type). 

@rdig Can a malformed event find its way into the block ingestor? I.e. an event without the expected set of arguments?

Reply:
Good to know, thanks 👍